server/controllers/gh.js

```diff
@@ -91,8 +115,53 @@ function makeIssueRequest(issueRequestUrl) {
         });
 }
 
+
+/**
+ * @param {String} issueRequestUrl required, url for API knocking
+ * @param {String} token required, token, generated in Passport to allow user change data
+ * @param {String} state default: 'closed', possible values 'closed' or 'open'
+ *
+ * @return {Promise} resolve - new state of issue, reject - some error
+ */
+function makeChangeIssueStateRequest(issueRequestUrl, token, state) {
+    logger.log('API request to', issueRequestUrl);
+
+    var possibleStates = ['closed', 'open'];
+    state || (state = 'closed');
+
+    if (typeof issueRequestUrl !== 'string' || typeof token !== 'string') {
+        return Promise.reject(new Error('Not enough required arguments'));
+    }
+
+    if (possibleStates.indexOf(state) === -1) {
+        return Promise.reject(new Error('Identifier "state" must be a string with value "closed" or "open"'));
+    }
+
+    return got(issueRequestUrl, {
+        method: 'PATCH',
+        json: true,
+        headers: {
+            'user-agent': 'bem-forum',
+            accept: 'application/vnd.github.v3.full+json',
+            authorization: 'token ' + token
+        },
+        body: JSON.stringify({
+            state: state
+        })
+    })
+    .then(
+        function (response) {
+            return response.body.state;
+        }
+    )
+    .catch(function (err) {
+        throw new Error(err.response.statusCode + ': ' + err.response.body.message);
```

Comment:
makeChangeIssueStateRequest - is a function that returned promise. Error from catch will be hoisted to promises caller in setIssueState. Is not it?